src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -61,196 +68,429 @@ public IActionResult OnPost(string provider, string returnUrl = null)
             return new ChallengeResult(provider, properties);
         }
 
-        public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, string remoteError = null)
+        public async Task<IActionResult> OnGetCallbackAsync(string providedReturnUrl = null, string remoteError = null)
```

Comment:
Found this a bit difficult to follow with all of the branching `return` statements, so not sure if that can be avoided, but nice work with the comments! 🙂 .

Reply:
Yeah, I did try to make it as painless as possible, but there's too many routes that we can take I think and each one requiring subtly different handling.  I figured that with the test coverage, we could hopefully avoid accidentally shooting ourselves in the foot when we make changes here.

Additionally I've raised https://dfedigital.atlassian.net/browse/EES-3165 in the backlog to imrove the coverage around the "invited users" code branches, as it's not as extensive as the "new user" code branches.